### DISCUSSION TITLE
Not able to import EPs from excel as Sample weight for BF method

### DISCUSSION THREAD

**Discussion by AmaanAliK** [casact/chainladder-python#453] [Q&A] [answered]
Hi guys,

I wanted to ask how I can best import the EPs from excel (its just one column) into the package as a sample weight for the BF method. It keeps giving me errors and I think its because Im unable to convert into the right format. My code is as below and thank you very much for the help as always!
`dfEP = pd.read_excel(file_path, sheet_name='EPy')
dfEP['Year'] = pd.to_datetime(dfEP['Year'], format='%Y%m')
raa = tri
sample_weight = dfEP
cl.BornhuetterFerguson(apriori=0.95).fit(
    X=raa, 
    sample_weight=sample_weight
).ibnr_.sum()`

**Comment by jbogaardt** [comment, marked as answer]
chainladder could really give more helpful error messages. That'd be a nice improvement,

Everything has to be a triangle object, including premium vectors:
```python
sample_weight = cl.Triangle(dfEP, origin='Year', columns="the_column_name_for_ep")
```

**Reply by AmaanAliK** [reply]
Thank you very much for your quick answer! I will give it a shot and post an update

**Reply by AmaanAliK** [reply]
Worked like a charm, thank you very much @jbogaardt :)
